### PR TITLE
Tenant chooser doesn't change the current tenant for the current view in Compass UI

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add `ApplicationContext` to `react-shared`
- Update `contextSwitcher` options to reflect current view
- Replace `builder` with `ApolloClientProvider`

**Related issue(s)**
[Compass](https://github.com/kyma-incubator/compass/pull/510)
[Bumps](https://github.com/kyma-project/kyma/pull/6458)
[CBS](https://github.com/kyma-project/kyma/pull/6452)

### CODE REVIEW

File: compass/src/config/luigi-config/helpers/navigation-helpers.js [hunk 1]
```diff
@@ -1,11 +1,32 @@
+const getAlternativePath = tenantUId => {
+  const currentPath = window.location.pathname;
+  const regex = new RegExp('^/tenant/(.*?)/(.*)(?:/.*)?');
+  const match = currentPath.match(regex);
+  if (match) {
+    const tenant = match[1];
+    const path = match[2];
+    if (tenant == tenantUId) {
+      // the same tenant, leave path as it is
+      return `${tenantUId}/${path}`;
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply compass/src/config/luigi-config/helpers/navigation-helpers.js]: `currentPath` contains _/tenant_ segment which is undesirable here.

File: shared/contexts/ApplicationContext.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      {children}
+    </ApplicationContext.Provider>
+  );
+};
+
+export function useApplicationContext() {
+  return useContext(ApplicationContext);
+}
+
+export const withApplicationContext = Component => props => (
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply shared/contexts/ApplicationContext.js]: I decided to left it here, as it's a shared component.

File: shared/contexts/ApplicationContext.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useContext, useState } from 'react';
+import LuigiClient from '@kyma-project/luigi-client';
+
+export const ApplicationContext = createContext({});
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply shared/contexts/ApplicationContext.js]: I decided to left it here, as it's a shared component.

File: compass/public-luigi/config/config.js [hunk 4]
```diff
@@ -1,5 +1,17 @@
...
+    {
+      name: 'default',
+      id: '3e64ebae-38b5-46a0-b1ed-9ccee153a0ae',
+    },
+    {
+      name: 'foo',
+      id: '1eba80dd-8ff6-54ee-be4d-77944d17b10b',
+    },
+    {
+      name: 'bar',
```
Comment (dariadomagala) [inline compass/public-luigi/config/config.js]: What if someone will configure it in a wrong way and set multiple tenants with same name? Should we somehow handle / not allow such behaviour?

File: compass/src/config/luigi-config/navigation.js [hunk 5]
```diff
@@ -29,11 +29,13 @@ const navigation = {
...
         {
           hideSideNav: true,
-          pathSegment: ':tenantId',
+          pathSegment: ':tenantName',
           navigationContext: 'tenant',
           context: {
             idToken: token,
-            tenantId: ':tenantId',
+            tenantName: ':tenantName',
+            defaultTenatnId: window.clusterConfig.defaultTenant,
```
Comment (dariadomagala) [inline compass/src/config/luigi-config/navigation.js]: defaultTena***nt***Id

**Pull request merged**